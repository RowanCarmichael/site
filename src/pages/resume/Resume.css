.resume {
  color: white;
  background: black;
  overflow-x: hidden;
}

.container {
  width: 100vw;
  background: black;
}

.background {
  height: 100%;
  left: 0;
  top: 0;
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.backgroundImage {
  object-fit: cover;
  width: 100vw;
  min-height: 100vh;
  animation: fade-in 0.4s;
}

.content {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 100%;
  height: 100vh;
  z-index: 1;
}

.header {
  font-size: 8rem;
  font-weight: 100;
  line-height: 8rem;
  text-align: center;
  margin: 0;
}

.resume h1 {
  font-size: 7rem;
  line-height: 7rem;
  font-weight: 100;
  margin: 0;
  width: 40rem;
  text-align: center;
  margin-top: auto;
  transform: rotate(270deg) translate(18rem,14rem);
}

.resume h2 {
  font-size: 2.8rem;
  margin: 0;
  font-weight: 100;
}

.resume li {
  font-size: 3rem;
  font-weight: bold;
}

.resume ul {
  list-style-type: circle;
}

.downloadResumeButton {
  font-size: 3rem;
  margin: 4rem;
  padding: 2.4rem;
  color: white;
}

.summaryContainer {
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex-wrap: wrap;
  margin: 4.8rem;
}

.summaryDetails div {
  margin-bottom: 2rem;
}

.experienceTitle {
  font-size: 3rem;
  font-weight: bold;
  margin: 0;
}

.experienceDetails {
  font-size: 2rem;
  margin: 0;
}

.arrow {
  fill: white;
  margin-left: 2.4rem;
  transform: scale(-1, 1) rotate(240deg);
  animation: bounce 1s infinite;
}

.downloadContainer {
  display: flex;
  cursor: pointer;
  outline: 0;
  margin: 0 4rem;
  align-items: center;
}

.final {
  height: calc(100vh - 8.6rem);
  width: 100vw;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes bounce {
  0% {
    margin-bottom: 0;
  }
  50% {
    margin-bottom: 2rem;
  }
}

@media only screen and (max-width: 1200px) {
  .resume h1 {
    transform: none;
    text-align: left;
  }

  .summaryContainer {
    flex-direction: column;
  }
}
