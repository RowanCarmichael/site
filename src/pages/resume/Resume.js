import React from 'react';
import { Container, Background } from 'react-fixed-background';
import Typewriter from '../../components/typewriter/Typewriter';
import styles from './Resume.css';
import PdfIcon from '../../assets/icons/pdf.svg';
import resume from '../../assets/files/RowanCarmichaelCV.pdf';
import FloatingButton from '../../components/buttons/FloatingButton';
import mountain from '../../assets/images/mountain.jpeg';
import desk from '../../assets/images/desk.jpeg';
import dock from '../../assets/images/dock.jpeg';
import keyboard from '../../assets/images/keyboard.jpeg';
import ArrowIcon from '../../assets/icons/arrow.svg';

class Resume extends React.Component {
  state = {
    loaded: false,
  }

  onLoadImage = () => {
    this.setState({ loaded: true });
  }

  downloadResume = () => {
    window.open(resume);
  }

  render() {
    return (
      <div className={styles.resume}>
        <FloatingButton onClick={this.downloadResume}>
          <PdfIcon width="70" height="70" />
        </FloatingButton>
        <Container className={styles.container}>
          <div className={styles.content}>
            <img onLoad={this.onLoadImage} className={this.state.loaded ? styles.backgroundImage : null} src={mountain} alt="mountain" />
          </div>
          <Background bringToFront className={styles.background}>
            <Typewriter speed={60} delay={1000} className={styles.header} text="See what I can do for you" />
          </Background>
        </Container>
        <Container className={styles.container}>
          <div className={styles.content}>
            <img className={styles.backgroundImage} src={desk} alt="desk" />
          </div>
          <Background bringToFront className={styles.background}>
            <div className={styles.summaryContainer}>
              <h1>SKILLS</h1>
              <div className={styles.summaryDetails}>
                <h2>Experienced In:</h2>
                <ul>
                  <li>ReactJS</li>
                  <li>JavaScript</li>
                  <li>HTML</li>
                  <li>CSS</li>
                </ul>
                <h2>Familiar With:</h2>
                <ul>
                  <li>SQL</li>
                  <li>Java</li>
                  <li>C#</li>
                  <li>Xamarin</li>
                </ul>
              </div>
            </div>
          </Background>
        </Container>
        <Container className={styles.container}>
          <div className={styles.content}>
            <img className={styles.backgroundImage} src={keyboard} alt="keyboard" />
          </div>
          <Background bringToFront className={styles.background}>
            <div className={styles.summaryContainer}>
              <h1>FAVOURITE LIBRARIES</h1>
              <div className={styles.summaryDetails}>
                <ul>
                  <li>Redux</li>
                  <li>Mobx</li>
                  <li>CSS Modules</li>
                  <li>Mocha</li>
                  <li>Eslint</li>
                  <li>Chai</li>
                  <li>Enzyme</li>
                  <li>Jest</li>
                  <li>Selenium</li>
                  <li>Flow</li>
                  <li>Babel</li>
                  <li>Webpack</li>
                  <li>Node.js (Express)</li>
                </ul>
              </div>
            </div>
          </Background>
        </Container>
        <Container className={styles.container}>
          <div className={styles.content}>
            <img className={styles.backgroundImage} src={dock} alt="dock" />
          </div>
          <Background bringToFront className={styles.background}>
            <div className={styles.summaryContainer}>
              <h1>EXPERIENCE</h1>
              <div className={styles.summaryDetails}>
                <div>
                  <p className={styles.experienceTitle}>React Frontend Engineer</p>
                  <p className={styles.experienceDetails}>EROAD Limited</p>
                  <p className={styles.experienceDetails}>Feb 2016 - Now</p>
                </div>
                <div>
                  <p className={styles.experienceTitle}>Wed Developer - Industry Project</p>
                  <p className={styles.experienceDetails}>OPUS International Consultants Limited</p>
                  <p className={styles.experienceDetails}>Mar 2015 - Nov 2015</p>
                </div>
                <div>
                  <p className={styles.experienceTitle}>Computer Science Lab Demonstrator</p>
                  <p className={styles.experienceDetails}>University of Auckland Computer Science Department</p>
                  <p className={styles.experienceDetails}>Mar 2015 - Nov 2015</p>
                </div>
                <div>
                  <p className={styles.experienceTitle}>BTech(Hons) Specialising in Information Technology</p>
                  <p className={styles.experienceDetails}>University of Auckland</p>
                  <p className={styles.experienceDetails}>Mar 2012 - Nov 2015</p>
                </div>
              </div>
            </div>
          </Background>
        </Container>
        <div className={styles.final}>
          <div className={styles.downloadContainer} onClick={this.downloadResume} role="button" tabIndex={0}>
            <h3 className={styles.header}>Download Full Resume</h3>
            <ArrowIcon className={styles.arrow} width="60" height="60" />
          </div>
        </div>
      </div>
    );
  }
}

export default Resume;
