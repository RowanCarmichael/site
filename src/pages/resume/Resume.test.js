import React from 'react';
import renderer from 'react-test-renderer';
import Resume from './Resume';

describe('Resume page', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<Resume />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
