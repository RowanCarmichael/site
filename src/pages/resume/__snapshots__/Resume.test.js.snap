// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume page renders correctly 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
    onClick={[Function]}
    role="button"
    tabIndex="0"
  >
    <pdf.svg
      height="70"
      width="70"
    />
  </div>
  <div
    className="Container__container___1UbtE"
  >
    <div
      className={undefined}
    >
      <img
        alt="mountain"
        className={null}
        onLoad={[Function]}
        src="mountain.jpeg"
      />
    </div>
    <div
      className="Background__backgroundOuter___2XEdG Background__foreground___2JmNj"
    >
      <div
        className="Background__backgroundInner___34S8e"
      >
        <div
          className={null}
          delay={1000}
          speed={60}
        >
          <span>
            
          </span>
          <span
            className={undefined}
          >
            |
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Container__container___1UbtE"
  >
    <div
      className={undefined}
    >
      <img
        alt="desk"
        className={undefined}
        src="desk.jpeg"
      />
    </div>
    <div
      className="Background__backgroundOuter___2XEdG Background__foreground___2JmNj"
    >
      <div
        className="Background__backgroundInner___34S8e"
      >
        <div
          className={undefined}
        >
          <h1>
            SKILLS
          </h1>
          <div
            className={undefined}
          >
            <h2>
              Experienced In:
            </h2>
            <ul>
              <li>
                ReactJS
              </li>
              <li>
                JavaScript (ES6+)
              </li>
              <li>
                HTML5
              </li>
              <li>
                CSS3 / LESS / Sass
              </li>
            </ul>
            <h2>
              Familiar With:
            </h2>
            <ul>
              <li>
                SQL
              </li>
              <li>
                Java
              </li>
              <li>
                C#
              </li>
              <li>
                Xamarin
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Container__container___1UbtE"
  >
    <div
      className={undefined}
    >
      <img
        alt="keyboard"
        className={undefined}
        src="keyboard.jpeg"
      />
    </div>
    <div
      className="Background__backgroundOuter___2XEdG Background__foreground___2JmNj"
    >
      <div
        className="Background__backgroundInner___34S8e"
      >
        <div
          className={undefined}
        >
          <h1>
            FAVOURITE LIBRARIES
          </h1>
          <div
            className={undefined}
          >
            <ul>
              <li>
                Redux
              </li>
              <li>
                Mobx
              </li>
              <li>
                CSS Modules
              </li>
              <li>
                Mocha
              </li>
              <li>
                Eslint
              </li>
              <li>
                Chai
              </li>
              <li>
                Enzyme
              </li>
              <li>
                Jest
              </li>
              <li>
                Selenium
              </li>
              <li>
                Flow
              </li>
              <li>
                Babel
              </li>
              <li>
                Webpack
              </li>
              <li>
                Node.js (Express)
              </li>
              <li>
                Yarn
              </li>
              <li>
                Npm
              </li>
              <li>
                Git
              </li>
              <li>
                Storybook
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Container__container___1UbtE"
  >
    <div
      className={undefined}
    >
      <img
        alt="dock"
        className={undefined}
        src="dock.jpeg"
      />
    </div>
    <div
      className="Background__backgroundOuter___2XEdG Background__foreground___2JmNj"
    >
      <div
        className="Background__backgroundInner___34S8e"
      >
        <div
          className={undefined}
        >
          <h1>
            EXPERIENCE
          </h1>
          <div
            className={undefined}
          >
            <div>
              <p
                className={undefined}
              >
                React Frontend Engineer
              </p>
              <p
                className={undefined}
              >
                EROAD Limited
              </p>
              <p
                className={undefined}
              >
                Feb 2016 - Now
              </p>
            </div>
            <div>
              <p
                className={undefined}
              >
                Wed Developer - Industry Project
              </p>
              <p
                className={undefined}
              >
                OPUS International Consultants Limited
              </p>
              <p
                className={undefined}
              >
                Mar 2015 - Nov 2015
              </p>
            </div>
            <div>
              <p
                className={undefined}
              >
                Computer Science Lab Demonstrator
              </p>
              <p
                className={undefined}
              >
                University of Auckland Computer Science Department
              </p>
              <p
                className={undefined}
              >
                Mar 2015 - Nov 2015
              </p>
            </div>
            <div>
              <p
                className={undefined}
              >
                BTech(Hons) Specialising in Information Technology
              </p>
              <p
                className={undefined}
              >
                University of Auckland
              </p>
              <p
                className={undefined}
              >
                Mar 2012 - Nov 2015
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <h3
        className={undefined}
      >
        Download Full Resume
      </h3>
      <arrow.svg
        className={undefined}
        height="60"
        width="60"
      />
    </div>
  </div>
</div>
`;
